test/lint/test_runner/src/main.rs

```diff
@@ -284,6 +283,47 @@ fn lint_doc() -> LintResult {
     }
 }
 
+// Find a program on $PATH
+fn which(program: &str) -> bool {
```

Comment:
My thought was to avoid running another subprocess which will then make the same syscalls (excluding checking it's executable).

`--version` is less code on our side though, and more robust, so I will switch to that.